File: 1221-element-appearing-more-than-25-in-sorted-array/element-appearing-more-than-25-in-sorted-array.py
class Solution(object):
    def findSpecialInteger(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        size = len(arr)
        qrt = size // 4
        ele = arr[0]
        count = 1
        for i in range(1, size):
            if arr[i] == ele:
                count+=1
            else:
                count = 1
            if count > qrt:
                return arr[i]
            ele = arr[i]
        return ele
        

File: 1221-element-appearing-more-than-25-in-sorted-array/element-appearing-more-than-25-in-sorted-array.cpp
class Solution {
public:
    int findSpecialInteger(vector<int>& arr) {
        int size = arr.size();
        int twenty = size/4;
        int count = 1;
        int ele = arr[0];
        for(int i = 1; i<size; i++){
            if(arr[i] == ele){
                count++;
            }
            else{
                count = 1;
            }
            if(count > twenty) return arr[i];
            ele = arr[i];
        }
        return ele;
    }
};